lib/commands/time.js

```diff
@@ -23,14 +23,31 @@ export async function getDeviceTime(format = MOMENT_FORMAT_ISO8601) {
 }
 
 /**
- * @this {import('../driver').AndroidDriver}
+ * @this {AndroidDriver}
  * @param {import('./types').DeviceTimeOpts} [opts={}]
  * @returns {Promise<string>}
  */
 export async function mobileGetDeviceTime(opts = {}) {
   return await this.getDeviceTime(opts.format);
 }
 
+/**
+ * @this {AndroidDriver}
+ * @param {string} zoneName
+ * @returns {Promise<void>}
+ */
+export async function adjustTimeZone(zoneName) {
+  if (!moment.tz.names().includes(zoneName)) {
+    throw new Error(
+      `The provided time zone identifier '${zoneName}' is not known. ` +
+      `Please choose a valid TZ identifier from https://en.wikipedia.org/wiki/List_of_tz_database_time_zones`
+    );
+  }
+  this.log.info(`Setting the device time zone to '${zoneName}'`);
+  await this.adb.shell(['service', 'call', 'alarm', '3', 's16', zoneName]);
```

Comment:
The `3` depends on the number of the method in `IAlarmManager.aidl` so the number could depends on Android OS versions. So I think it would be nice to give API version as an argument in case we need to modify the number.

https://cs.android.com/android/platform/superproject/+/master:frameworks/base/apex/jobscheduler/framework/java/android/app/IAlarmManager.aidl;l=1?q=IAlarmManager


I haven't checked any cases, but for example when https://android.googlesource.com/platform/frameworks/base/+/f76a50c/core/java/android/app/IAlarmManager.aidl case, the number might be `5`. (randomly googled)

Reply:
never mined for the api level as an argument. we could call `const apiLevel = await this.adb.getApiLevel()`.

Maybe adding comment the `3` could be another number by depending on the aidl for the API level?

Reply:
added a comment